### ISSUE TITLE
Wrong number of arguments error incorrectly says functions take zero arguments

### ISSUE THREAD

**Issue by laniakea64** [casey/just#2230]
Using `just` 1.30.0.  If a function takes a fixed non-zero number of arguments, but is called with the wrong number of arguments, the error message claims the function takes zero arguments, regardless of the actual correct number.

Example invalid justfile -
```
foo := datetime()
```

```
$ just --evaluate
error: Function `datetime` called with 0 arguments but takes 0
 ——▶ justfile:1:8
  │
1 │ foo := datetime()
  │        ^^^^^^^^
```

Same issue with too many arguments -
```
foo := env_var('FOO', 'invalid')
```

```
$ just --evaluate 
error: Function `env_var` called with 2 arguments but takes 0
 ——▶ justfile:1:8
  │
1 │ foo := env_var('FOO', 'invalid')
  │        ^^^^^^^
```

Seems to only affect functions that take a single fixed number of arguments.  Functions that can take different numbers of arguments give the expected error message here.

**Comment by casey** [comment]
Nice catch! Appreciate the report. I think this has been broken for a long time, at least it predates just 1.30.0. I was using exclusive ranges in a somewhat nonsensical way. Fixed in #2231.

**Comment by casey** [comment]
Released 1.30.1 with the fix. It's not a big bug, but I think it's confusing enough to be worth fixing in a patch release.
